lib/legacy.coffee

```diff
@@ -51,15 +52,21 @@ module.exports = (ClassicLinter) ->
         suffix: textEditor.getGrammar().scopeName
       }
       return new Promise((resolve, reject) ->
-        temp.open(tmpOptions, (err, info) ->
+        temp.mkdir('AtomLinter', (err, tmpDir) ->
           return reject(err) if err
 
           try
-            fs.writeSync(info.fd, textEditor.getText())
-            fs.closeSync(info.fd)
+            tmpFile = path.join(tmpDir, path.basename(filePath))
+            fs.writeFileSync(tmpFile, textEditor.getText())
```

Comment:
I hate Node's callback style because it makes the code really ugly and awkward to work with.

It's supposed to stop the event loop. That's what `*Sync` means.